operator/pkg/controllers/etcd/defaultspec.go

```diff
@@ -0,0 +1,190 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package etcd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/awslabs/kit/operator/pkg/apis/infrastructure/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	defaultEtcdReplicas = 3
+	defaultEtcdImage    = "public.ecr.aws/eks-distro/etcd-io/etcd:v3.4.14-eks-1-18-1"
+)
+
+func (e *Provider) DefaultStaticSpecFor(controlPlane *v1alpha1.ControlPlane) func() *v1alpha1.ETCDSpec {
+	return func() *v1alpha1.ETCDSpec {
+		return &v1alpha1.ETCDSpec{
+			Spec: &v1.PodSpec{
```

Comment:
Consider separating the default pod spec from everything else. I think it'll clean up the defaulting logic in the future if you add top level fields to etcd spec.